packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -51,11 +51,30 @@ class ChannelCommand extends FlutterCommand {
     }
   }
 
-  Future<void> _listChannels({ bool showAll, bool verbose }) async {
+  String _formatChannels(String currentBranch, String channel) =>
+      channel == currentBranch ? '* $channel' : '  $channel';
+
+  /// This method sorts the officialChannels by stability.
+  ///
+  /// It needs to check that the length of the list is the correct one
+  /// because it may receive an incomplete array, since is used as an accumulator.
+  List<String> _sortOfficialChannelsByStability(List<String> officialChannels) {
+    if (officialChannels.length == FlutterVersion.officialChannels.length) {
+      final List<String> orderedByStabilityChannels = List<String>.filled(FlutterVersion.officialChannels.length, '');
```

Comment:
It's kind of weird to initialize the list with empty strings. maybe something like this:

final List<String> result = <String>[]..length = FlutterVersion.officialChannels.length;
for (String channel in officialChannels) {
  final int index = FlutterVersion.officialChannels[channel]
  result[index] = channel;
}

Reply:
Instead of:
```dart
final List result = []..length = FlutterVersion.officialChannels.length;
``` 
which gives me troubles with type annotation, i opt for: 
```dart
final List<String> result = List<String>(FlutterVersion.officialChannels.length);
```

*Out of curiosity*, what's the syntax to specify the type of List literals in Dart? I wasn't able to find it.